File: CSharp-Part-2/6. Strings and Text Processing/StringsAndTextProcessing/Palindromes/Palindromes.cs
// Write a program that extracts from a given text all palindromes, e.g. ABBA, lamal, exe.

using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Linq;

class Palindromes
{
    static void Main()
    {
        string text = "Hello. This is an example that contains palindromes.exe and I like listening to ABBA, but have no idea what lamal is.";

        StringBuilder outputStr = new StringBuilder();
        foreach (var match in Regex.Matches(text, @"\w+"))
        {
            string word = match.ToString(); 
            if (word.SequenceEqual(word.Reverse())) // && word.Length > 1 in the condition if one-letter words are not considered palindromes.
            {
                outputStr.AppendLine(match.ToString());
            }
        }
        Console.WriteLine("List of palindromes: ");
        Console.Write(outputStr);
    }
}

File: JavaScript-Fundamentals/11. Strings/palindromes.js
// Write a program that extracts from a given text all palindromes, e.g. "ABBA", "lamal", "exe".

var text = "ABBA is the best band, ever. God knows what a lamal is. Window's executables have an exe extension.";

function checkPalindrome(str) {
	if (str.length < 2) {
		return false;
	}

	var tmp = str.split('').reverse();
	return tmp.join('') === str;
}

function extractPalindromes(text) {
	var result = [];

	text.split(' ').forEach(function (word) {
		if (checkPalindrome(word)) {
			result.push(word);
		}
	});

	return result;
}

console.log(extractPalindromes(text));